packages/neptune-extension/src/common/state/configuration/reducer.ts

```diff
@@ -0,0 +1,46 @@
+import {API_TOKEN_LOCAL_STORAGE_KEY, ApiTokenParsed, parseApiToken} from 'common/api/auth';
+import {ConfigurationActions} from "./actions";
+
+interface ConfigurationState {
+  apiToken?: string
+  apiTokenParsed?: ApiTokenParsed
+  isApiTokenValid?: boolean
+  inferredUsername?: string
+}
+
+const initialApiToken = localStorage.getItem(API_TOKEN_LOCAL_STORAGE_KEY) || undefined;
+const initialApiTokenParsed = initialApiToken ? parseApiToken(initialApiToken) : undefined;
+
+const initialState: ConfigurationState = {
+  apiToken: initialApiToken,
+  apiTokenParsed: initialApiTokenParsed,
+};
+
+export function configurationReducer(state: ConfigurationState = initialState, action:ConfigurationActions): ConfigurationState {
+  switch (action.type) {
+    case "API_TOKEN_SET": {
+      return {
+        ...state,
+        apiToken: action.payload.token,
+        apiTokenParsed: parseApiToken(action.payload.token),
```

Comment:
It looks like in any case when the function is called, the token is already parsed, so there should be no need to parse it here again.